File: LeetCode/[E] - 242. Valid Anagram/solution.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        sMap = {}
        for char in s:
            sMap[char] = sMap.get(char, 0) + 1
        for char in t:
            if char not in sMap:
                return False
            sMap[char] -= 1
        for cnt in sMap.values():
            if cnt < 0 or cnt > 0:
                return False
        return True

File: LeetCode/[E] - 242. Valid Anagram/solution.go
package main

func isAnagram(s string, t string) bool {
	sMap := make(map[rune]int)
	for _, r := range s {
		sMap[r] += 1
	}
	for _, r := range t {
		if _, ok := sMap[r]; !ok {
			return false
		}
		sMap[r]--
	}
	for _, value := range sMap {
		if value < 0 || value > 0 {
			return false
		}
	}
	return true
}
